src/marqo/tensor_search/models/search.py

```diff
@@ -59,13 +60,42 @@ def add_content(self, content: List[Union[str, List[str]]]) -> VectorisedJobPoin
             end_idx=len(self.content)
         )
 
+
 class SearchContextTensor(BaseModel):
     vector: List[float]
     weight: float
 
 
+class SearchContextDocumentsParameters(BaseModel):
+    tensorFields: Optional[List[str]] = None
+    excludeInputDocuments: bool = True
+    concurrency: Optional[int] = None
+
+    @validator('tensorFields', pre=True, always=True)
+    def check_tensor_fields_not_empty(cls, v):
+        if v == []:
+            raise InvalidArgError('context document tensorFields parameter must be non-empty list.'
```

Comment:
Raise `ValueError` in the pydantic validator

Reply:
done